apps/docs/content/guides/auth/auth-anonymous.mdx

```diff
@@ -0,0 +1,130 @@
+```
+
+</TabPanel>
+</Tabs>
+
+## Convert the anonymous user to a permanent user
+
+Converting an anonymous user to a permanent user requires linking an identity to the user.
+
+### Link an Email / Phone identity
+
+<Tabs
+  scrollable
+  size="small"
+  type="underlined"
+  defaultActiveId="js"
+  queryGroup="language"
+>
+<TabPanel id="js" label="JavaScript">
+
+You can use the [`updateUser()`](/docs/reference/javascript/auth-updateuser) method to link an email or phone identity to the anonymous user.
+
+```js
+const { data, error } = await supabase.auth.updateUser({ email: 'example@email.com' })
+```
+
+</TabPanel>
+</Tabs>
+
+### Link an OAuth identity
+
+<Tabs
+  scrollable
+  size="small"
+  type="underlined"
+  defaultActiveId="js"
+  queryGroup="language"
+>
+<TabPanel id="js" label="JavaScript">
+
+You can use the [`linkIdentity()`](/docs/reference/javascript/auth-linkidentity) method to link an oauth identity to the anonymous user.
+
+```js
+const { data, error } = await supabase.auth.linkIdentity({ provider: 'google' })
+```
+
+</TabPanel>
+</Tabs>
+
+## Access control
+
+An anonymous user assumes the `authenticated` role just like a permanent user. You can use row-level security (RLS) policies to differentiate between an anonymous user and a permanent user by checking for the `is_anonymous` claim in the JWT returned by `auth.jwt()`:
+
+```sql
+create policy "Only permanent users can post to the news feed"
+on news_feed for insert
+to authenticated
+with check ( auth.jwt()->>'is_anonymous' = false );
+
+create policy "Anonymous and permanent users can view the news feed"
+on news_feed for select
+to authenticated
+using ( true );
+
+```
+
+## Conflict Resolution
+
+Depending on your application requirements, data conflicts can arise when an anonymous user is converted to a permanent user. For example, in the context of an e-commerce application, an anonymous user would be allowed to add items to the shopping cart without signing up / signing in. When they decide to sign-in to an existing account, you will need to decide how you want to resolve data conflicts in the shopping cart:
+
+1. Overwrite the items in the cart with those in the existing account
+2. Overwrite the items in the cart with those from the anonymous user
+3. Merge the items in the cart together
+
+## Abuse prevention and rate limits
+
+Since anonymous users are stored in your database, bad actors can abuse the endpoint to increase your database size drastically. It is strongly recommended to [enable silent captcha or turnstile](/dashboard/project/_/settings/auth) to prevent abuse for anonymous sign-ins. There is also an IP-based rate limit for anonymous sign-ins at 30 requests per hour which can be configured in your [dashboard](/dashboard/project/_/auth/rate-limits). You can refer to the full list of rate limits [here](/docs/guides/platform/going-into-prod#rate-limiting-resource-allocation--abuse-prevention).
```

Comment:
```suggestion
Since anonymous users are stored in your database, bad actors can abuse the endpoint to increase your database size drastically. It is strongly recommended to [enable silent captcha or turnstile](/docs/guides/auth/auth-captcha) to prevent abuse for anonymous sign-ins. There is also an IP-based rate limit for anonymous sign-ins at 30 requests per hour which can be configured in your [dashboard](/dashboard/project/_/auth/rate-limits). You can refer to the full list of rate limits [here](/docs/guides/platform/going-into-prod#rate-limiting-resource-allocation--abuse-prevention).
```